src/internal/template/template.go

```diff
@@ -87,33 +87,35 @@ func (values Values) Apply(component types.ZarfComponent, path string) {
 		"GIT_AUTH_PULL":      values.secret.gitPull,
 	}
 
-	// Don't template component-specifric variables for every component
+	// Don't template component-specific variables for every component
 	switch component.Name {
 	case "zarf-agent":
-		mappings["AGENT_CRT"] = base64.StdEncoding.EncodeToString(values.agentTLS.Cert)
-		mappings["AGENT_KEY"] = base64.StdEncoding.EncodeToString(values.agentTLS.Key)
-		mappings["AGENT_CA"] = base64.StdEncoding.EncodeToString(values.agentTLS.CA)
+		builtinMap["AGENT_CRT"] = base64.StdEncoding.EncodeToString(values.agentTLS.Cert)
+		builtinMap["AGENT_KEY"] = base64.StdEncoding.EncodeToString(values.agentTLS.Key)
+		builtinMap["AGENT_CA"] = base64.StdEncoding.EncodeToString(values.agentTLS.CA)
 
 	case "zarf-seed-registry", "zarf-registry":
-		mappings["SEED_REGISTRY"] = values.seedRegistry
-		mappings["HTPASSWD"] = values.secret.htpasswd
-		mappings["REGISTRY_SECRET"] = values.secret.registrySecret
+		builtinMap["SEED_REGISTRY"] = values.seedRegistry
+		builtinMap["HTPASSWD"] = values.secret.htpasswd
+		builtinMap["REGISTRY_SECRET"] = values.secret.registrySecret
 
 	case "logging":
-		mappings["LOGGING_AUTH"] = values.secret.logging
-
+		builtinMap["LOGGING_AUTH"] = values.secret.logging
 	}
 
 	// Iterate over any custom variables and add them to the mappings for templating
-	for key, value := range component.Variables {
-		mappings[key] = value
+	templateMap := map[string]string{}
+	for key, value := range builtinMap {
+		// Builtin keys are always uppercase in the format ###ZARF_KEY###
+		templateMap[strings.ToUpper(fmt.Sprintf("###ZARF_%s###", key))] = value
 	}
 
-	message.Debug(mappings)
-
-	for template, value := range mappings {
-		// Keys are always uppercase in the format ###ZARF_KEY###
-		template = strings.ToUpper(fmt.Sprintf("###ZARF_%s###", template))
-		utils.ReplaceText(path, template, value)
+	for key, value := range config.VariableMap {
+		// Variable keys are always uppercase in the format ###ZARF_VAR_KEY###
+		templateMap[strings.ToUpper(fmt.Sprintf("###ZARF_VAR_%s###", key))] = value
 	}
+
+	message.Debug(templateMap)
```

Comment:
Can we also log the `builtinMap` like we originally were here before the variable name change?

Also, I personally think it's better to log the variableNames (or some type of added context) that way it's easier to figure out what something is when going through the logs
`message.Debugf("templateMap = %#v", templateMap)`

Reply:
The template map contains the builtinmap. Would you want to print the same info twice? (agree with `message.Debugf("templateMap = %#v", templateMap)`

Reply:
Oh whoops I missed that `templateMap` contained `builtinMap`. Thanks!